tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -198,16 +198,32 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
         }
 
       case ModuleKind.ESModule =>
-        foreachImportedModule { (module, pos0) =>
+        foreachImportedModule { (module, pos0, path) =>
           implicit val pos = pos0
           val from = js.StringLiteral(module)
           val moduleBinding = jsGen.envModuleField(module).ident
-          val importStat = js.ImportNamespace(moduleBinding, from)
-          builder.addJSTree(importStat)
+          path match {
+            // namespace import or default import
+            case Nil | List("default") =>
+              val importStat = js.ImportNamespace(moduleBinding, from)
+              builder.addJSTree(importStat)
+            case _ =>
+              val bindings = path.map(p => (js.ExportName(p), js.Ident(p)))
```

Comment:
On the same line, but unrelated: simply using `js.Ident(p)` is not safe, as `p` could collide with one of the identifiers used internally by the emitter for other purposes. For example `$uD` or `$c_jl_Class`. You'll have to use appropriate avoidance techniques such as those already present in `jsEnv.envModuleField`.

You could also have the same member imported from two different modules, which would also collide using the current code. Therefore the name of the identifiers should contain the encoded module name as prefix or something like that (always taking care that they really cannot collide with anything else).